astropy/cosmology/core.py

```diff
@@ -210,6 +210,9 @@ def __init__(self, name=None, meta=None):
         all_vars["name"].__set__(self, name)
         all_vars["meta"].__set__(self, OrderedDict(meta or {}))
 
+    def __post_init__(self):  # noqa: B027
```

Comment:
"`Cosmology.__post_init__` is an empty method in an abstract base class, but has no abstract decorator Ruff[B027](https://beta.ruff.rs/docs/rules/empty-method-without-abstract-decorator)"

Reply:
``super(). __post_init__ `` errors if there's not ``__post_init__`` method so it's necessary to have the empty method. And thus the noqa.

Reply:
But why are the subclasses calling `super().__post_init__()` if it doesn't do anything?

Reply:
Because they get mixed in with `FlatFLRWMixin` to make the flat cosmologies. Both have ``__post_init__`` so it's necessary to ensure stuff passes up the chain.